src/item.h

```diff
@@ -670,6 +677,45 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      */
     int chip_resistance( bool worst = false ) const;
 
+    /** How much damage has the item sustained? */
+    int damage() const {
+        return damage_;
+    }
+
+    /** Minimum amount of damage to an item (state of maximum repair) */
+    int min_damage() const;
+
+    /** Maximum amount of damage to an item (state before destroyed) */
+    int max_damage() const;
+
+    /**
+     * Apply damage to item constrained by @ref min_damage and @ref max_damage
+     * @param qty maximum amount by which to adjust damage (negative permissible)
+     * @param dmg type of damage which may be passed to @ref on_damage callback
+     * @return whether item should be destroyed
+     */
+    bool mod_damage( int qty, damage_type dt = DT_NULL );
```

Comment:
> // Just cap damage in the getter

And in `mod_damage` plus any other functions we later implement. `damage_` shouldn't have a magic value that means `destroyed`. At some point that will crop up as an error.

> The bool return value also makes sense as it matches the current way of handling item damage in the code

I agree it's idiomatic with the current codebase

> check whether the item was destroyed and remove the object in one known place

Assign the result to a variable `bool destroyed = inc_damage()` and later check `if( destroyed )` which would be synonymous with `if( is_destroyed() )` except without imposing the extra complexity in circumstances where it's not required.

> I'm not sure about the damage type argument, though. What would it be used for?

The inspiration for this PR was that I want vehicle parts that take damage to acquire faults - some of which are more probable for certain types of damage. For example bashing damage could damage an air filter or belt but electrical damage probably wouldn't.